examples/query_string.rs

```diff
@@ -13,23 +16,42 @@ async fn main() {
     pretty_env_logger::init();
 
     // get /example1?key=value
+    // demonstrates an optional parameter.
     let example1 = warp::get()
         .and(warp::path("example1"))
         .and(warp::query::<HashMap<String, String>>())
         .map(|p: HashMap<String, String>| match p.get("key") {
             Some(key) => Response::builder().body(format!("key = {}", key)),
-            None => Response::builder().body(String::from("No param in query.")),
+            None => Response::builder().body(String::from("No \"key\" param in query.")),
         });
 
     // get /example2?key1=value,key2=42
+    // uses the query string to populate a custom object
     let example2 = warp::get()
         .and(warp::path("example2"))
         .and(warp::query::<MyObject>())
         .map(|p: MyObject| {
             Response::builder().body(format!("key1 = {}, key2 = {}", p.key1, p.key2))
         });
 
-    warp::serve(example1.or(example2))
+    // get /example3?key1=value,key2=42
+    // builds on example2 but adds custom error handling
+    let example3 =
+        warp::get()
+            .and(warp::path("example3"))
+            .and(warp::query::<MyObject>().map(Some).or_else(|_| async {
```

Comment:
can we isolate the optional query param, so that it's easier to read?

Reply:
Sure thing, I have submitted an update.